src/shogun/base/zip_iterator.h

```diff
@@ -0,0 +1,107 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Gil Hoben
+ */
+
+#ifndef SHOGUN_ZIP_ITERATOR_H
+#define SHOGUN_ZIP_ITERATOR_H
+
+#include <tuple>
+#include <utility>
+
+namespace shogun
+{
+	template <typename... Args1, typename... Args2, size_t... Idx>
```

Comment:
I can't figure out: does this thing check that the two iterators have the same "size" somehow? Or does that even make sense?

Reply:
no, but i can add a runtime check that could do that (would need know what method to call). this checks that all the values of the current iteration are equals to another iteration, i.e. `iter == container.end()`, where `iter` would be a tuple of pointers